File: kotlin/src/main/kotlin/dao/TaskDao.kt
package dao

import org.springframework.stereotype.Repository
import model.Task
import java.util.LinkedHashMap
import java.util.concurrent.atomic.AtomicInteger

Repository
public class TaskDao {

    val tasks: LinkedHashMap<Int,Task> = LinkedHashMap()
    var lastid: AtomicInteger = AtomicInteger(0)

    public fun findAll(): Collection<Task> {
        return tasks.values()
    }

    public fun findById(id: Int): Task? {
        return tasks.get(id)
    }

    public fun save(task: Task) {
        if (task.id == null) {
            task.id = lastid.incrementAndGet()
        }
        tasks.put(task.id!!, task)
    }
}

File: scala/src/main/scala/dao/TaskDao.scala
package dao

import org.springframework.stereotype.Repository
import model.Task
import collection.mutable
import java.util.concurrent.atomic.AtomicInteger

@Repository
class TaskDao {

  val tasks = new mutable.LinkedHashMap[Int,Task]
  var lastid = new AtomicInteger

  def findAll(): Iterable[Task] = {
    tasks.values
  }

  def findById(id: Int): Option[Task] = {
    tasks.get(id)
  }

  def save(task: Task) {
    if (task.id == 0) {
      task.id = lastid.incrementAndGet
    }
    tasks.put(task.id, task)
  }
}